Initial basic functionality of short form

rraub opened the pull request:
This should be the bare minimum to get the basic "ask a question" short form working. Nothing glamorous yet, just enough to get shipped and we can work on improving it as we go.

This plugin adds a shortcode `[asu-rfi-form]` and a settings page for defining the `source_id` for the side.

Todo:
- [x] client side validation of fields
- [x] cache control settings
- [x] site wide settings for source_id


![screen shot 2016-12-12 at 3 38 20 pm](https://cloud.githubusercontent.com/assets/295804/21119802/6a034bc2-c081-11e6-8d86-c5d55b0efc9c.png)

Success Banner:
![screen shot 2016-12-12 at 3 38 35 pm](https://cloud.githubusercontent.com/assets/295804/21119803/6a1b5802-c081-11e6-8cad-f68d76ac8039.png)

Error Banner:
![screen shot 2016-12-12 at 3 42 14 pm](https://cloud.githubusercontent.com/assets/295804/21119834/951a8384-c081-11e6-9f35-6d228c794be6.png)

Settings Menu Location:
![screen shot 2016-12-13 at 12 43 24 pm](https://cloud.githubusercontent.com/assets/295804/21156084/c728ccae-c131-11e6-8e0f-7cbc1a6e3db6.png)

rraub commented:
@nathan-rollins I dont quite like the wording of the question "I am going to be a.." but having some optional section for type of student is nice. Do you have any ideas for a better phrase?

nathan-rollins commented on src/services/asu-degree-service.php:
```diff
@@ -0,0 +1,98 @@
+   *  "2177" for Summer, 2017
+   */
+  public static function get_available_enrollment_terms() {
+    $semseter_names = array(
```
FYI: spelling typo

nathan-rollins commented on src/views/rfi-form/form.handlebars:
```diff
@@ -0,0 +1,161 @@
+        </div>
+      </div>
+      <div class="form-group has-feedback">
+        <label class="control-label col-sm-3" for="submit"></label>
```
Is this label useful for accessibility/screen readers? Maybe an alternate aria attribute could be leveraged to make the submit button obvious (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/Basic_form_hints).

nathan-rollins commented on src/views/rfi-form/simple-request-info-form.handlebars:
```diff
@@ -0,0 +1,73 @@
+      
+      
+      <div class="form-group has-feedback">
+        <label class="control-label col-sm-3" for="submit"></label>
```
Same possible issue with accessibility for this label.

rraub commented:
I've replaced the empty label for the submit button with an aria-label; it is kinda redundant but its more explicit for some older screen readers. (Much better than an empty label) Also I've added `aria-required="true"` for screen readers that don't yet support html 5.
